openstack/blockstorage/extensions/volumeactions/requests.go

```diff
@@ -88,6 +112,23 @@ func Detach(client *gophercloud.ServiceClient, id string, opts DetachOptsBuilder
 	return
 }
 
+// UploadImage will upload image base on the values in UploadImageOptsBuilder
+func UploadImage(client *gophercloud.ServiceClient, id string, opts UploadImageOptsBuilder) UploadImageResult {
```

Comment:
Let's group all the parts of this operation together (i.e. this function and  `UploadImageOpts` and its methods)